### PR TITLE
Allow java extensions for source files

### LINKED ISSUE #83: Make groovyc wrapper uses joint compilation
Currently the groovy language server only works with `.groovy` files, but we should set the jointCompilation configuration in compilationConfiguration to make it work with Java too.
https://github.com/apache/groovy/blob/master/src/main/org/codehaus/groovy/control/CompilerConfiguration.java#L155

### PR DESCRIPTION
**Pull request by natgabb**
Closes #83 
- Refactor GroovycWrapperTest to no longer compare sizes of expected vs expected results

### CODE REVIEW

File: groovy-language-server/src/test/java/com/palantir/ls/groovy/compilation/GroovycWrapperTest.java [hunk 1]
```diff
@@ -174,6 +174,27 @@ public void testCompile() throws InterruptedException, ExecutionException, IOExc
...
+                "class Coordinates {\n"
+                        + "   double latitude\n"
+                        + "   double longitude\n"
+                        + "   double getAt(int idx) {\n"
+                        + "      if (idx == 0) latitude\n"
+                        + "      else if (idx == 1) longitude\n"
+                        + "      else throw new ExceptionNew(\"Wrong coordinate index, use 0 or 1\")\n"
+                        + "   }\n"
+                        + "}\n");
+        addFileToFolder(root.getRoot(), "ExceptionNew.java", "public class ExceptionNew {}");
```
Comment (jiahuijiang) [inline groovy-language-server/src/test/java/com/palantir/ls/groovy/compilation/GroovycWrapperTest.java:187]: let's add some extra tests:
1. inside the java file only define a new class ExceptionOld
2. only have a java file where the syntax will fail even when evaluating with groovy syntax

Comment (natgabb) [reply groovy-language-server/src/test/java/com/palantir/ls/groovy/compilation/GroovycWrapperTest.java:187]: done

**Comment by natgabb** [comment]
@JiahuiJiang any more comments? :)

File: groovy-language-server/src/main/java/com/palantir/ls/groovy/compilation/GroovyWorkspaceCompiler.java [hunk 2]
```diff
@@ -201,10 +201,13 @@ public void handleChangeWatchedFiles(List<? extends FileEvent> changes) {
     }
 
     private void addAllSourcesToCompilationUnit() {
+        Set<String> allowedExtensions =
```
Comment (jiahuijiang) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/compilation/GroovyWorkspaceCompiler.java]: We can move this `allowedExtensions` also to `GroovyConstants`

**Comment by jiahuijiang** [comment]
+1 after a minor comment

File: groovy-language-server/src/main/java/com/palantir/ls/groovy/compilation/GroovyWorkspaceCompiler.java (hunk 2 above)
Comment (natgabb) [reply groovy-language-server/src/main/java/com/palantir/ls/groovy/compilation/GroovyWorkspaceCompiler.java]: done

**Pull request merged**